diff -ruN cmocka-1.1.5/meson.build cmocka-1.1.5-fix/meson.build
--- cmocka-1.1.5/meson.build	2022-11-14 22:29:23.000000000 -0500
+++ cmocka-1.1.5-fix/meson.build	2022-11-14 21:52:10.000000000 -0500
@@ -85,6 +85,12 @@
     )
     # We can't test the build type, so we can' add -D_FORTIFY_SOURCE=2 here
     if machine.system() == 'darwin'
+      macosx_version_min = '-mmacosx-version-min=10.8'
+      add_project_arguments(macosx_version_min, language: ['c'])
+      add_project_link_arguments(macosx_version_min, language: ['c'])
+      arch = ['-arch', 'x86_64', '-arch', 'arm64']
+      add_project_arguments(arch, language: ['c'])
+      add_project_link_arguments(arch, language: ['c'])
       if compiler.has_argument('-Wno-deprecated-declarations')
         add_project_arguments('-Wno-deprecated-declarations', language : ['c'], native: is_native)
       endif
